src/UKHO.ERPFacade.Common/Operations/IO/Azure/AzureTableReaderWriter.cs

```diff
@@ -64,11 +64,11 @@ public async Task UpdateEntityAsync<TKey, TValue>(string partitionKey, string ro
             }
         }
 
-        public IList<TableEntity> GetAllEntities(string partitionKey)
+        public async Task<IList<TableEntity>> GetEntitiesByQueryParameterAsync<TKey, TValue>(KeyValuePair<TKey, TValue> parameter)
         {
             var records = new List<TableEntity>();
```

Comment:
query returns list so no need to create another list to return. Please check once